### PR TITLE
Add `Reduce` method in reverse_iter

### PR DESCRIPTION
**Pull request by jeon-yongjun**
add `reduce` method to reverse_iter and obliterated the failure that was foreseen by it

### CODE REVIEW

File: vm/src/builtins/dict.rs [hunk 1]
```diff
@@ -837,6 +837,20 @@ macro_rules! dict_view {
...
             }
+            #[pymethod(magic)]
+            fn reduce(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyTupleRef {
+                let iter = builtins_reversed(vm).to_owned();
+                let internal = zelf.internal.lock();
+                // TODO: entries must be reversed too
+                let entries = match &internal.status {
+                    IterStatus::Active(dict) => dict
+                        .into_iter()
+                        .map(|(key, value)| ($result_fn)(vm, key, value))
```
Comment (youknowone) [inline vm/src/builtins/dict.rs:848]: here is a clippy error.

**Review by youknowone — CHANGES_REQUESTED** [review]
This patch also fixed a test  `test_enum::test_unique_composite`. Please remove expectedFailure decorator from the test.

**Comment by Snowapril** [comment]
> This patch also fixed a test `test_enum::test_unique_composite`. Please remove expectedFailure decorator from the test.

It seems temporary error. we cant replicate our environment(window, ubuntu).
We add `expectedFailure` again and it works now 😊

**Review by youknowone — APPROVED** [review]
Looks good, thank you for contributing!

**Pull request merged**